internal/tiltfile/tiltextension/plugin_test.go

```diff
@@ -191,6 +195,311 @@ include('Tiltfile.b')
+v1alpha1.extension(name='nested_real', repo_name='custom', repo_path='nested/real')
+
+load("ext://nested/fake", "printFake")
+printFake()
+
+load("ext://nested/real", "printReal")
+printReal()
+`, f.tmp.Path()))
+
+	fakeContent := `
+def printFake():
+    print("fake")
+	`
+
+	realContent := `
+def printReal():
+	print("real")
+	`
+
+	f.tmp.WriteFile(filepath.Join("my-custom-repo", "fake", "Tiltfile"), fakeContent)
+	f.tmp.WriteFile(filepath.Join("my-custom-repo", "nested", "real", "Tiltfile"), realContent)
+
+	res := f.assertExecOutput("fake\nreal")
+	f.assertLoadRecorded(res, "nested/fake", "nested/real")
+}
+
+func TestRepoLoadHost(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// We don't want to have to bother with file:// escaping on windows.
+		// The repo reconciler already tests this.
+		t.Skip()
+	}
+
+	// Assert that extension repositories with a load_host allow "autoregistration" of extensions if
+	// the extension path starts with the registered repository load_host.
+	f := newExtensionFixture(t)
+
+	f.tiltfile(fmt.Sprintf(`
+v1alpha1.extension_repo(name='custom', url='file://%s/ext-repo', load_host='custom')
+
+load("ext://custom/ext", "printFoo")
+printFoo()
+`, f.tmp.Path()))
+
+	f.tmp.WriteFile(filepath.Join("ext-repo", "ext", "Tiltfile"), libText)
+
+	res := f.assertExecOutput("foo")
+	f.assertLoadRecorded(res, "custom/ext")
+}
+
+func TestRepoGitSubpath(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// We don't want to have to bother with file:// escaping on windows.
+		// The repo reconciler already tests this.
+		t.Skip()
+	}
+
+	t.Skip("cannot use file:// repos with git_subpath, making this test unusable")
+
+	// Assert that extension repositories with a defined subpath load registered extensions
+	// from that subpath
+	f := newExtensionFixture(t)
+
+	f.tiltfile(fmt.Sprintf(`
+v1alpha1.extension_repo(name='custom', url='file://%s/ext-repo', git_subpath='subdir')
+v1alpha1.extension(name='my-ext', repo_name='custom')
+v1alpha1.extension(name='my-ext-with-path', repo_name='custom', repo_path='subdir2')
+
+# Assert that loading an extension without a repo_path loads from the repo-wide path
+load("ext://my-ext", "printExt")
+printExt()
+
+load("ext://my-ext-with-path", "printExt2")
+printExt2()
+`, f.tmp.Path()))
+
+	extContent := `
+def printExt():
+	print("main ext")
+	`
+
+	extContent2 := `
+def printExt2():
+	print("sub ext")
+	`
+
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "Tiltfile"), extContent)
+	f.tmp.WriteFile(filepath.Join("ext-repo", "subdir", "subdir2", "Tiltfile"), extContent2)
+
+	res := f.assertExecOutput("main ext\nsub ext")
+	f.assertLoadRecorded(res, "my-ext", "my-ext-with-path")
+}
+
+func TestRepoLoadHostAndSubpath(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// We don't want to have to bother with file:// escaping on windows.
+		// The repo reconciler already tests this.
+		t.Skip()
+	}
+
+	t.Skip("cannot use file:// repos with git_subpath, making this test unusable")
```

Comment:
Due to the validation when registering an extension repository we cannot actually run this test after the change to block `file://` repos with a `git_subpath` test.

I don't know if there's a simple way to support `https://` repos in the test without requiring the network.